### DISCUSSION TITLE
🗳️ Grails 7 Scope - What should the Java Baseline for Grails 7 be?

### DISCUSSION THREAD

**Discussion by sdelamo** [apache/grails-core#13534] [Polls]
Since Spring Boot 3.0, the baseline version is Java 17. As we plan to support the latest spring boot, I think we need to raise to a minimum of Java 17. Maybe we could take the opportunity to go to 21 as the Java baseline.

**Comment by matrei** [comment]
What would we gain from raising the baseline from 17 to 21?
(You can already run Grails 6 on Java 22)

Enabling virtual threads is to my understanding just a Spring Boot setting if you are running the app with Java 21+.
`spring.threads.virtual.enabled=true`

I think we should follow the baseline of the Spring Boot version used.
Spring Boot 3.3 is baselined on Java 17.

(https://github.com/grails/grails-core/tree/7.0.x is already on Java 17)

**Reply by gsartori** [reply]
I would follow Spring Boot dependencies versions as well, adding Gradle to the list of matching versions.

**Comment by sdelamo** [comment]
We are going in the [17 as Java baseline](https://github.com/grails/grails-core/pull/13532) direction.
